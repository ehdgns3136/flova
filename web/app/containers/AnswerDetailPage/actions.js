/*
 *
 * AnswerDetailPage actions
 *
 */
import { normalize } from 'normalizr';
import {
  ANSWER_GET_REQUEST,
  ANSWER_GET_SUCCESS,
  ANSWER_GET_FAILURE,
  QUESTION_GET_REQUEST,
  QUESTION_GET_SUCCESS,
  QUESTION_GET_FAILURE,
  INITIALIZE_REDUCER,
} from './constants';

import { answerSchema, questionSchema } from '../../global/entities/schemas';

export function answerGetRequestAction(answerID) {
  return {
    type: ANSWER_GET_REQUEST,
    answerID,
  };
}

export function answerGetSuccessAction(answer) {
  return {
    type: ANSWER_GET_SUCCESS,
    payload: normalize(answer, answerSchema),
  };
}

export function answerGetFailureAction() {
  return {
    type: ANSWER_GET_FAILURE,
  };
}

export function questionGetRequestAction(answerID) {
  return {
    type: QUESTION_GET_REQUEST,
    answerID,
  };
}

export function questionGetSuccessAction(question) {
  return {
    type: QUESTION_GET_SUCCESS,
    payload: normalize(question, questionSchema),
  };
}

export function questionGetFailureAction() {
  return {
    type: QUESTION_GET_FAILURE,
  };
}

export function initializeReducer() {
  return {
    type: INITIALIZE_REDUCER,
  };
}
