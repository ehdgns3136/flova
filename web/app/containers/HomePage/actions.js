import { normalize } from 'normalizr';
import {
  activitiesSchema,
} from 'global/entities/schemas';
import {
  CLEAR_FEED,
  CHANGE_ACTIVE_CATEGORY,
  CHECK_ACTIVE_CATEGORY,
  GET_ACTIVITIES_REQUEST,
  GET_ACTIVITIES_SUCCESS,
  GET_ACTIVITIES_FAILURE,
} from './constants';


export function clearFeed() {
  return {
    type: CLEAR_FEED,
  };
}

export function getActivitiesRequest(params) {
  return {
    type: GET_ACTIVITIES_REQUEST,
    params,
  };
}

export function getActivitiesSuccess(activities, nextCursor, previousCursor) {
  return {
    type: GET_ACTIVITIES_SUCCESS,
    payload: normalize(activities, activitiesSchema),
    nextCursor,
    previousCursor,
  };
}

export function getActivitiesFailure() {
  return {
    type: GET_ACTIVITIES_FAILURE,
  };
}

export function changeActiveCategory(category) {
  return {
    type: CHANGE_ACTIVE_CATEGORY,
    category,
  };
}

export function checkActiveCategory(params) {
  return {
    type: CHECK_ACTIVE_CATEGORY,
    params,
  };
}
