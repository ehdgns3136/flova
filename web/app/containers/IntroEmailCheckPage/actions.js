/*
 *
 * IntroPage actions
 *
 */

import {
  CHECK_EMAIL_EXIST_REQUEST,
  CHECK_EMAIL_EXIST_SUCCESS,
  CHECK_EMAIL_EXIST_FAILURE,
} from './constants';

export function checkEmailExistRequestAction(email) {
  return {
    type: CHECK_EMAIL_EXIST_REQUEST,
    email,
  };
}

export function checkEmailExistSuccessAction() {
  return {
    type: CHECK_EMAIL_EXIST_SUCCESS,
  };
}

export function checkEmailExistFailureAction() {
  return {
    type: CHECK_EMAIL_EXIST_FAILURE,
  };
}
