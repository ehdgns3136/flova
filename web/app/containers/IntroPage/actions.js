/*
 *
 * IntroPage actions
 *
 */

import {
  LOGIN_REQUEST,
  LOGIN_SUCCESS,
  LOGIN_FAILURE,
  SOCIAL_USER_CONTINUE,
  CHECK_SOCIAL_USER_EXIST_REQUEST,
  STORE_TOKEN_REQUEST,
  EMPTY_TOKEN_REQUEST,
} from './constants';

export function loginRequestAction(email, password) {
  return {
    type: LOGIN_REQUEST,
    email,
    password,
  };
}

export function loginSuccessAction() {
  return {
    type: LOGIN_SUCCESS,
  };
}

export function loginFailureAction() {
  return {
    type: LOGIN_FAILURE,
  };
}

export function socialUserContinueRequestAction(token, tokenType) {
  return {
    type: SOCIAL_USER_CONTINUE,
    token,
    tokenType,
  };
}

export function checkSocialUserExistRequestAction(token, tokenType) {
  return {
    type: CHECK_SOCIAL_USER_EXIST_REQUEST,
    token,
    tokenType,
  };
}

export function storeTokenRequestAction(token, tokenType) {
  return {
    type: STORE_TOKEN_REQUEST,
    token,
    tokenType,
  };
}

export function emptyTokenRequestAction() {
  return {
    type: EMPTY_TOKEN_REQUEST,
  };
}
