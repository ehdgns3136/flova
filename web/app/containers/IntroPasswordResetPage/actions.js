/*
 *
 * IntroPasswordResetPage actions
 *
 */

import {
  VALIDATE_PASSWORD_RESET_KEY_REQUEST,
  VALIDATE_PASSWORD_RESET_KEY_SUCCESS,
  VALIDATE_PASSWORD_RESET_KEY_FAILURE,
  PASSWORD_RESET_REQUEST,
  PASSWORD_RESET_SUCCESS,
  PASSWORD_RESET_FAILURE,
} from './constants';

export function validatePasswordResetKeyRequestAction(key) {
  return {
    type: VALIDATE_PASSWORD_RESET_KEY_REQUEST,
    key,
  };
}

export function validatePasswordResetKeySuccessAction(email, key) {
  return {
    type: VALIDATE_PASSWORD_RESET_KEY_SUCCESS,
    email,
    key,
  };
}

export function validatePasswordResetKeyFailureAction() {
  return {
    type: VALIDATE_PASSWORD_RESET_KEY_FAILURE,
  };
}

export function passwordResetRequestAction(password, key) {
  return {
    type: PASSWORD_RESET_REQUEST,
    password,
    key,
  };
}

export function passwordResetSuccessAction() {
  return {
    type: PASSWORD_RESET_SUCCESS,
  };
}

export function passwordResetFailureAction() {
  return {
    type: PASSWORD_RESET_FAILURE,
  };
}
