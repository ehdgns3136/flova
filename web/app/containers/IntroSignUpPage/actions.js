import {
  SIGN_UP_REQUEST,
  SIGN_UP_SUCCESS,
  SIGN_UP_FAILURE,
  MOVE_TO_TOPIC,
  MOVE_TO_SIGN_UP,
  MOVE_TO_TERMS,
  LOAD_TOPICS_REQUEST,
  LOAD_TOPICS_SUCCESS,
  SELECT_TOPIC_ITEM,
  ADD_TOPICS_TO_USER_REQUEST,
  SIGN_UP_SUCCESS_REDIRECT_REQUEST,
  SET_TOPICS_EMPTY,
} from './constants';

export function signUpRequestAction(email, password, name) {
  return {
    type: SIGN_UP_REQUEST,
    email,
    password,
    name,
  };
}

export function signUpSuccessAction() {
  return {
    type: SIGN_UP_SUCCESS,
  };
}

export function signUpFailureAction(errorMessage) {
  return {
    type: SIGN_UP_FAILURE,
    errorMessage,
  };
}

export function moveToTopicAction() {
  return {
    type: MOVE_TO_TOPIC,
  };
}

export function moveToSignUpAction() {
  return {
    type: MOVE_TO_SIGN_UP,
  };
}

export function moveToTermsAction() {
  return {
    type: MOVE_TO_TERMS,
  };
}

export function loadTopicsRequestAction() {
  return {
    type: LOAD_TOPICS_REQUEST,
  };
}

export function loadTopicsSuccessAction(topicList) {
  return {
    type: LOAD_TOPICS_SUCCESS,
    topicList,
  };
}

export function selectTopicItemAction(topicID) {
  return {
    type: SELECT_TOPIC_ITEM,
    topicID,
  };
}

export function addTopicsToUserRequestAction(selectedTopicItemList) {
  return {
    type: ADD_TOPICS_TO_USER_REQUEST,
    selectedTopicItemList,
  };
}


export function signUpSuccessRedirectRequestAction() {
  return {
    type: SIGN_UP_SUCCESS_REDIRECT_REQUEST,
  };
}

export function setTopicsEmptyAction() {
  return {
    type: SET_TOPICS_EMPTY,
  };
}
