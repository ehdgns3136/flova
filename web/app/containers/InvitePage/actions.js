/*
 *
 * IntroPage actions
 *
 */

import {
  VALIDATE_INVITE_KEY_FAILURE,
  VALIDATE_INVITE_KEY_REQUEST,
  VALIDATE_INVITE_KEY_SUCCESS,
} from './constants';

export function validateInviteKeyRequest(inviteKey) {
  return {
    type: VALIDATE_INVITE_KEY_REQUEST,
    inviteKey,
  };
}

export function validateInviteKeySuccess(payload) {
  return {
    type: VALIDATE_INVITE_KEY_SUCCESS,
    payload,
  };
}

export function validateInviteKeyFailure(error) {
  return {
    type: VALIDATE_INVITE_KEY_FAILURE,
    error,
  };
}
