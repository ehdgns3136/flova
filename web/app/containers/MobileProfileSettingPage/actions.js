/*
 *
 * MobileProfileSettingPage actions
 *
 */

import {
  GET_PROFILE_FAILURE,
  GET_PROFILE_REQUEST,
  GET_PROFILE_SUCCESS,
} from './constants';

export function getProfileRequest(id) {
  return {
    type: GET_PROFILE_REQUEST,
    id,
  };
}

export function getProfileSuccess(profile) {
  return {
    type: GET_PROFILE_SUCCESS,
    profile,
  };
}

export function getProfileFailure() {
  return {
    type: GET_PROFILE_FAILURE,
  };
}
