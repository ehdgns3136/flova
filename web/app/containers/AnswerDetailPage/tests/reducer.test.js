
import { fromJS, is } from 'immutable';
import answerDetailPageReducer, { initialState } from '../reducer';
import { initializeReducer } from '../actions';

describe('answerDetailPageReducer', () => {
  it('returns the initial state', () => {
    expect(answerDetailPageReducer(undefined, {})).toMatchSnapshot();
  });

  it('should initialize reducer', () => {
    expect(is(answerDetailPageReducer(undefined, initializeReducer()), initialState)).toBe(true);

    const MOCK_STATE = fromJS({
      answer: null,
      question: null,
      isAnswersLoading: true,
      isQuestionLoading: true,
    });

    expect(is(answerDetailPageReducer(MOCK_STATE, initializeReducer()), initialState)).toBe(true);
  });
});
