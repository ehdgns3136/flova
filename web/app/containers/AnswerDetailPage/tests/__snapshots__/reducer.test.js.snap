exports[`answerDetailPageReducer returns the initial state 1`] = `
Object {
  "answer": null,
  "isAnswerLoading": false,
  "isQuestionLoading": false,
  "question": null,
}
`;
